scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -780,6 +781,26 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
     private def maybeParamType(allowFunctionType: Boolean): Type =
       if (allowFunctionType) paramType() else typ()
 
+    private def namedTypeOpt(
+        fmods: => List[Mod],
+        allowFunctionType: Boolean
+    ): Option[Type.TypedParam] = currToken match {
+      case t: Ident if tryAhead[Colon] =>
+        val startPos = prevIndex
+        next() // skip colon
+        val mods = fmods
+        val modPos = if (mods.isEmpty) startPos else mods.head.begIndex
+        val name = atPos(startPos)(Type.Name(t.value))
+        val tpe = maybeParamType(allowFunctionType)
+        Some(autoEndPos(modPos)(Type.TypedParam(name, tpe, mods)))
+      case _ => None
+    }
+
+    private def maybeNamedType(): Type = {
+      val ok = dialect.allowImprovedTypeClassesSyntax
```

Comment:
That's not really related to the change. That's named tuples, but we didn't add a separate dialect for it.

Reply:
didn't realize that. so named tuples are ok to be parsed for any dialect, correct? ok, and other than, what do you think about this alternative?